storagemarket/integration_test.go

```diff
@@ -277,65 +281,103 @@ func TestMakeDealNonBlocking(t *testing.T) {
 	// Provider should be blocking on waiting for funds to appear on chain
 	pd := providerDeals[0]
 	assert.Equal(t, result.ProposalCid, pd.ProposalCid)
-	shared_testutil.AssertDealState(t, storagemarket.StorageDealProviderFunding, pd.State)
+	require.Eventually(t, func() bool {
+		return pd.State == storagemarket.StorageDealProviderFunding
+	}, 1*time.Second, 100*time.Millisecond, "actual deal status is %s", storagemarket.DealStates[pd.State])
 }
 
+// FIXME Gets hung sometimes
 func TestRestartClient(t *testing.T) {
```

Comment:
@hannahhoward Some test cases here can fail in hairy edge cases. I've documented those edge cases for each test as comments on the test case.

Please let me know what you think about handling them. If they seem rare enough, we can just let it be and avoid the complexity.